org.eclipse.buildship.ui/src/main/java/org/eclipse/buildship/ui/internal/project/GradleProjectConfigurator.java

```diff
@@ -0,0 +1,101 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Christoph Läubrich.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ ******************************************************************************/
+package org.eclipse.buildship.ui.internal.project;
+
+import java.io.File;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+
+import org.eclipse.buildship.core.BuildConfiguration;
+import org.eclipse.buildship.core.GradleBuild;
+import org.eclipse.buildship.core.GradleCore;
+import org.eclipse.buildship.core.internal.configuration.GradleProjectNature;
+import org.eclipse.buildship.core.internal.workspace.NewProjectHandler;
+import org.eclipse.buildship.core.internal.workspace.SynchronizationJob;
+import org.eclipse.core.resources.IContainer;
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.runtime.IPath;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.ui.wizards.datatransfer.ProjectConfigurator;
+
+
+public class GradleProjectConfigurator implements ProjectConfigurator {
+
+  @Override
+  public Set<File> findConfigurableLocations(File root, IProgressMonitor monitor) {
+    if(isGradleRoot(root)) {
+      return Set.of(root);
+    }
+    return Set.of();
+  }
+
+  @Override
+  public void removeDirtyDirectories(Map<File, List<ProjectConfigurator>> proposals) {
+    Set<File> gradleRoots = new HashSet<>();
+    Set<File> nonGradleRoots = new HashSet<>();
+    for(Entry<File, List<ProjectConfigurator>> entry : proposals.entrySet()) {
+      File root = entry.getKey();
+      if(isGradleRoot(root)) {
+        entry.getValue().removeIf(pc -> !(pc instanceof GradleProjectConfigurator));
+        gradleRoots.add(root);
+      } else {
+        nonGradleRoots.add(root);
+      }
+    }
+    if(!gradleRoots.isEmpty()) {
+      for(File file : nonGradleRoots) {
+        proposals.remove(file);
+      }
+    }
+  }
+
+  private boolean isGradleRoot(File root) {
+    if(root != null) {
+      if(new File(root, "gradle.properties").isFile()) {
```

Comment:
its recommend best practice to use a [`settings` file](https://docs.gradle.org/current/userguide/organizing_gradle_projects.html#sec:settings_file). 
`gradle.properties` is not required, although it is a reasonable indicator for a project.
I would suggest to at least also check for `settings.gradle` and `settings.gradle.kts`.